### PR TITLE
[wip] Feature/timestamp dbal test

### PR DESCRIPTION
**Pull request by TomHAnderson**

### CODE REVIEW

**Comment by TomHAnderson** [comment]
This is where I'm at now.  I added the DBAL test and the whole thing went to bits.  The kicker is all the database tests pass!  But the broken tests don't seem to be related to the DBAL test.  I'm going to sleep on this.  Any idea you have would be welcome.

File: src/Illuminate/Database/DBAL/TimestampType.php [hunk 1]
```diff
@@ -36,7 +36,9 @@ public function getSQLDeclaration(array $fieldDeclaration, AbstractPlatform $pla
                 return $this->getSQLitePlatformSQLDeclaration($fieldDeclaration);
 
             default:
-                throw new DBALException('Invalid platform: '.$name);
+                throw new RuntimeException(
+                    'Registering a custom Doctrine type requires Doctrine DBAL (doctrine/dbal).'
+                );
```
Comment (bakerkretzmar) [inline src/Illuminate/Database/DBAL/TimestampType.php:41]: I think the original except and message here were correct—if we hit this point and need to throw an exception, the issue is not that Doctrine isn't installed, it's that we appear to be trying to use a database platform that isn't supported or doesn't exist, right?

File: tests/Database/DBAL/TimestampTypeTest.php [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    {
+        Schema::create('test', function (Blueprint $table) {
+            $table->addColumn('datetime', 'datetime_to_timestamp');
+        });
+
+        Schema::table('test', function (Blueprint $table) {
+            $table->timestamp('datetime_to_timestamp')->nullable(true)->change();
+        });
+
+        $this->assertTrue(Schema::hasColumn('test', 'datetime_to_timestamp'));
```
Comment (bakerkretzmar) [inline tests/Database/DBAL/TimestampTypeTest.php:40]: Let's also verify here that the column is now actually of type `timestamp` in the database

Comment (TomHAnderson) [reply tests/Database/DBAL/TimestampTypeTest.php:40]: Agreed, but I didn't know how to do that.  Have you got a solution?

File: src/Illuminate/Database/DBAL/TimestampType.php (hunk 1 above)
Comment (TomHAnderson) [reply src/Illuminate/Database/DBAL/TimestampType.php:41]: The previous exception didn't *exist* at all.  So I replaced it with a RuntimeException.

File: src/Illuminate/Database/DBAL/TimestampType.php [hunk 3]
```diff
@@ -36,7 +36,9 @@ public function getSQLDeclaration(array $fieldDeclaration, AbstractPlatform $pla
                 return $this->getSQLitePlatformSQLDeclaration($fieldDeclaration);
 
             default:
-                throw new DBALException('Invalid platform: '.$name);
+                throw new RuntimeException(
```
Comment (TomHAnderson) [inline src/Illuminate/Database/DBAL/TimestampType.php:39]: The DBALException class didn't exist.  The RuntimeException is appropriate.  I've reverted the message.

**Comment by TomHAnderson** [comment]
Moved to #9

File: src/Illuminate/Database/DBAL/TimestampType.php (hunk 3 above)
Comment (bakerkretzmar) [reply src/Illuminate/Database/DBAL/TimestampType.php:39]: @TomHAnderson what about using `Doctrine\DBAL\Exception`? It looks like that actually even has a method specifically for this type of error.

File: tests/Database/DBAL/TimestampTypeTest.php (hunk 2 above)
Comment (bakerkretzmar) [reply tests/Database/DBAL/TimestampTypeTest.php:40]: I'll work on this, I've done it before but it's ugly 😅

**Pull request closed without merging**